web/components/InputEdgeConfig.jsx

```diff
@@ -0,0 +1,239 @@
+			<Tag color="error">Inconsistent</Tag>
+		</Tooltip>;
+	}
+	if (target_edge.target_edge !== edge.id) {
+		// eslint-disable-next-line max-len
+		return <Tooltip title={`Edge on target instance does not have this edge as its target (${target_edge.target_edge})`}>
+			<Tag color="error">Inconsistent</Tag>
+		</Tooltip>;
+	}
+
+	return <Tag color="success">OK</Tag>;
+}
+
+export function InputEdgeConfig({ value, onChange }) {
+	// Shim for clusterio being inconsistent with object vs stringified object handling
+	if (typeof value === "string") {
+		value = JSON.parse(value);
+	}
+
+	const [visible, setVisible] = React.useState(false);
+	const [newValue, setNewValue] = React.useState(value);
+
+	const edges = newValue.edges || [];
+
+	function displayEdge(edge, index) {
+		return <div key={`${index} ${edge.edge?.id}`}>
+			<Divider orientation="right">
+				<Space>
+					<EdgeStatusTag
+						edge={edge}
+					// Clusterio does not currently provide a way for a config input field to access the current
+					// instances ID so I am leaving this code disabled
+					// instanceId={instanceId}
+					/>
+					<Button onClick={() => {
+						setNewValue({
+							...newValue, edges: [
+								...edges.slice(0, index),
+								...edges.slice(index + 1),
+							],
+						});
+					}}>
+						<DeleteOutlined />
+					</Button>
+				</Space>
+			</Divider>
+			<EditEdge
+				key={index}
+				edge={edge}
+				onChange={(newEdge) => {
+					setNewValue({
+						...newValue, edges: [
+							...edges.slice(0, index),
+							newEdge,
+							...edges.slice(index + 1),
+						],
+					});
+				}} />
+		</div>;
+	}
+
+	return <>
+		<Button onClick={() => {
+			setNewValue(value);
+			setVisible(true);
+		}}>
+			Configure Edges
+		</Button>
+		<Modal
+			title="Edge Configuration"
+			open={visible}
+			onOk={() => {
+				onChange(JSON.stringify(newValue));
+				setVisible(false);
+			}}
+			onCancel={() => {
+				setVisible(false);
+			}}
+		>
+			{edges.map(displayEdge)}
+			{/* Button to add new edge */}
+			<Button onClick={() => {
+				setNewValue({
+					...newValue, edges: [
+						...edges,
+						{
+							surface: 1,
+							origin: [0, 0],
+							direction: 0,
+							length: 10,
+						},
+					],
+				});
+			}}>
+				<PlusOutlined /> Add Edge
+			</Button>
+		</Modal>
+	</>;
+}
+
+function EditEdge({ edge, onChange }) {
+	const leftProps = {
+		style: {
+			width: "30%",
+			display: "inline-block",
+			verticalAlign: "middle",
+		},
+	};
+	// Fields to edit edge properties
+	return <div>
+		<div>
+			<span {...leftProps}>Edge ID</span>
+			<InputNumber
+				value={edge.id}
+				onChange={(value) => onChange({ ...edge, id: value })}
+			/>
+		</div>
+		<div>
+			<span {...leftProps}>Origin position</span>
+			<InputNumber
+				value={edge.origin?.[0]}
+				formatter={(value) => `x ${value}`}
+				parser={value => value.replace("x ", "")}
+				onChange={(value) => onChange({ ...edge, origin: [value, edge.origin?.[1]] })}
```

Comment:
It never had an origin before I defaulted it to have one. It can still not have one if defined manually using the cli for example. This pattern is required to input the array properly. If we want to avoid undefined to end up in the config like that we need a validation guard on the top level save function, not here.